extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,972 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+1. [State](#state): A document-scope, mutable JSON state. State variables update in response to user actions. `amp-bind` does not evaluate expressions on page load. Visual elements should have their default "state" defined and not rely `amp-bind` for initial render.
+2. [Expressions](#expressions): JavaScript-like expressions that can reference the **state**.
+3. [Bindings](#bindings): Special attributes that link an element's property to a **state** via an **expression**. A property is bound by wrapping it inside brackets, in the form of `[property]`.
+
+### Basic example
+
+[example preview="inline" playground="true" imports="amp-bind"]
 
 ```html
 <p [text]="'Hello ' + foo">Hello World</p>
 
-<button on="tap:AMP.setState({foo: 'amp-bind'})">Say "Hello amp-bind"</button>
+<button on="tap:AMP.setState({foo: 'Interactivity'})">
+  Say "Hello Interactivity"
+</button>
 ```
 
-{% call callout('Note', type='note') %}
-For performance and to avoid the risk of unexpected content jumping, `amp-bind` does not evaluate expressions on page load. This means that the visual elements should be given a default state and not rely `amp-bind` for initial render.
-{% endcall %}
+[/example]
 
-### How does it work?
+In the example above:
 
-`amp-bind` has three main components:
+- The **state** begins as empty.
+- It has a single **binding** to `[text]`, the text content of a node, on the `<p>` element.
+- The `[text]` value contains the **expression**, `'Hello ' + foo`. This expression concatenates the string 'Hello ' and the value of the **state variable** foo.
 
-1. [State](#state): A document-scope, mutable JSON state. In the example above, the state is empty before tapping the button. After tapping the button, the state is `{foo: 'amp-bind'}`.
-2. [Expressions](#expressions): These are JavaScript-like expressions that can reference the **state**. The example above has a single expression, `'Hello ' + foo`, which concatenates the string literal `'Hello '` and the state variable `foo`.
-   There is a limit of 100 operands what can be used in an expression.
-3. [Bindings](#bindings): These are special attributes of the form `[property]` that link an element's property to an **expression**. The example above has a single binding, `[text]`, which updates the `<p>` element's text every time the expression's value changes.
+When the user taps/clicks the button:
 
-`amp-bind` takes special care to ensure speed, security and performance on AMP pages.
+1. It triggers the `tap` event.
+1. The `tap` event invokes the `AMP.setState()` method.
+1. The `AMP.setState()` methods sets the `foo` **state variable** to the value of `Interactivity`.
+1. The state is no longer empty, so the page updates the bound property to its state.
+
+[tip type="note"]
+Calling `AMP.setState()` in some examples may set or change states of other examples on page. Refresh this page to see examples before `AMP.setState()`.
```

Comment:
Possible for us to iframe each example so they don't share a single global state? Or use different values in each example?

Reply:
We're going to work through the examples in the next iteration. Its just too much right now. Thanks for the suggestions!